src/prefect/deployments/flow_runs.py

```diff
@@ -192,3 +196,174 @@ async def run_deployment(
             await anyio.sleep(poll_interval)
 
     return flow_run
+
+
+@async_dispatch(arun_deployment)
+def run_deployment(
+    name: Union[str, UUID],
+    client: Optional["SyncPrefectClient"] = None,
```

Comment:
While this is decorated with `@async_dispatch`, I think we might need to only accept `PrefectClient` instances. My thinking is that if a `SyncPrefectClient` is passed in an async context, then `arun_deployment` will fail. 

The transition will be a little messy, but I think we can create a sync client internally if one isn't passed and handle a provided async client with `run_coro_as_sync`. Once we start warning on async dispatch, we can also start warning on passed async clients so that users know to provide a sync client or use `arun_deployment`.

Reply:
ah that's a great point

Reply:
hmm the tricky thing about this is that run_coro_as_sync might try to put it in a new thread, which causes "bound to a different even loop" issues 🤔

Reply:
What if we did the run in the current loop or `asyncio.run` trick? I'm not sure if we have a utility for that, but it might be useful for this migration.